server/main_server/src/rodsServer.cpp

```diff
@@ -718,13 +718,26 @@ namespace
                 }
             }
 
-            char hostname[HOST_NAME_MAX]{};
+            char hostname[HOST_NAME_MAX + 1]{};
             if (gethostname(hostname, sizeof(hostname)) != 0) {
                 log_server::error("Failed to retrieve local server's hostname for delay server migration.");
                 return;
             }
+
             log_server::trace("Local server's hostname = [{}]", hostname);
 
+            if (const auto hn = resolve_hostname(hostname, hostname_resolution_scheme::match_preferred); hn) {
+                log_server::trace("Local server's hostname resolved via host_resolution configuration = [{}]", *hn);
+
+                if (hn->size() >= sizeof(hostname)) {
+                    log_server::error(
+                        "Local server's resolved hostname exceeds [{}] bytes (i.e. HOST_NAME_MAX).", HOST_NAME_MAX);
+                    return;
+                }
+
+                std::strncpy(hostname, hn->c_str(), sizeof(hostname));
+            }
```

Comment:
This logic is incorrect.

Reply:
Adjusted the implementation to resolve the local, leader, and successor host names against the host_resolution server_config.json property.

Verified at that the server will use the preferred hostnames for migration if defined.